### DISCUSSION TITLE
subdomains

### DISCUSSION THREAD

**Discussion by alexiokay** [nuxt/nuxt#19685] [Questions]
### Describe the feature

I'am looking for possibility to implement subdomains for my Nuxt3 project. I was looking for solution, but I havent found anything for the 3 version. Maybe somebody did it already?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [x] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by FloryanFilip** [comment]
@danielroe I recall reading somewhere in nuxt2 related issue also regarding subdomains that you said it should be supported natively with nuxt3. Any hints as to how?

**Reply by CeruttiMaicon** [reply]
I would also find it useful, especially for projects that work with multi tenants

**Comment by edeuxk** [comment]
@alexiokay I was looking to access to subdomain in a middleware.
Here is the trick to know the host and then parse the subdomain using split('.')[0]

```
export default defineNuxtRouteMiddleware((to, from) => {
  const event = useRequestEvent();
  console.log(event.node.req.headers.host);
});
```

**Comment by RyanPaulGannon** [comment]
Also would like to be able to use subdomains, would be very willing to help but unsure where to start ...

**Comment by juandelperal** [comment]
Same here, would be a great addition to support subdomains natively

**Comment by RyanPaulGannon** [comment]
I found this, https://twitter.com/atinux/status/1657812230816014338?s=46&t=XGiV21reetVKTt6j4CydQA

I haven't been able to try it in my own project just yet (I'm also trying to figure it out with Azure), but hopefully it could help someone

**Reply by Chuiantw1212** [reply]
Keypress alike approach is borken with most recent nuxt3 version.
With Nuxt3.8.1
![image](https://github.com/nuxt/nuxt/assets/10844438/4db9c7a1-cb30-4d11-b76d-bd426378873e)

Keypress(Nuxt rc.11)
![image](https://github.com/nuxt/nuxt/assets/10844438/1311f085-c790-4201-aad6-ecaee6e62d2b)

**Reply by danielroe** [reply]
It's not broken as such; the generated path now has a pair of brackets at the end. `/test/:id()`

**Reply by Chuiantw1212** [reply]
Thanks for the reply.
Knowing that the pair of the brackets are not the problem,
I'm gonna try tomorrow to check which version is good for the Keypress approch.

**Reply by desk-pro** [reply]
@Chuiantw1212 Have you found a solution ? I have the same problem you mentionned. 
![image](https://github.com/nuxt/nuxt/assets/89840442/d14578de-56f6-4bfe-a167-36d22cab20d9)

**Comment by pitambar-jha** [comment]
I have been using subdomain and multi domain support in production with nuxt 3. Have explained it here in this thread: https://github.com/nuxt/nuxt/discussions/16142#discussioncomment-6484723.  There are other ways of handling this depending on the use case, in my case i felt this was the cheapest solution which fulfilled the security and performance needs.
